### ISSUE TITLE
"Standalone" doc2html bat and js don't work on NT 5.1: "can't create object", or "object doesn't support this property or method"

### ISSUE THREAD

**Issue by wesinator** [ildar-shaimordanov/jsxt#4]
Hi, very interesting framework and tools here

I did notice that the doc2html bat available in the releases, and the doc2html.js file don't seem to work in Windows XP hosts:
![jsxt_doc2html_error_winxp_object_doesnt_support_this_property](https://github.com/user-attachments/assets/92f4886f-ad9e-4247-b9b1-40a0acba7378)

on the bat, when given `/F:HTML` and any .doc file, the code returns the error `Error encountered: automation server can't create object`.
In the usage with the .js in WScript, the usage page of the script ran without arguments doesn't provide any info. With a .doc file argument, the script gives an error `Error encountered: object doesn't support this property or method`.

[NO_TRAIN]::

**Comment by ildar-shaimordanov** [comment]
Hi @wesinator 

Thanks for your interest to my work. Recently, I don't have any computer with MS Office installed within. I need time to investigate it and understand what is wrong with the script.

The only thing I can say that the script initially was developed under Windows XP with MS Office on board and it was working that time.

Sorry for asking a silly question but I'd like to be sure in it. Is MS Word from MS Office installed on your XP host?

**Comment by wesinator** [comment]
> Hi @wesinator
> 
> Thanks for your interest to my work. Recently, I don't have any computer with MS Office installed within. I need time to investigate it and understand what is wrong with the script.
> 
> The only thing I can say that the script initially was developed under Windows XP with MS Office on board and it was working that time.
> 
> Sorry for asking a silly question but I'd like to be sure in it. Is MS Word from MS Office installed on your XP host?

Hi Ildar, thanks for your reply and the work you've done on this unique tool.

Office was not installed - some old .doc format files were on device and could be opened with Wordpad.

I wasn't aware from the docs or info that Office is required for the doc2html to work.

[NO_TRAIN]::

**Comment by ildar-shaimordanov** [comment]
@wesinator, no worries. It's certainly my fault. i forgot to mention about necessity MS Office on board. Later I will do this and close the issue.

**Comment by wesinator** [comment]
would it be possible to make it work without Office?
I'm also curious what APIs or COM controls it uses that are from Office.

[NO_TRAIN]::

**Comment by wesinator** [comment]
Thanks
I wonder if OpenOffice has an ActiveX control like `Word.Application`? (or if there is a basic Word OLE activex that Wordpad uses)

[NO_TRAIN]::

**Comment by ildar-shaimordanov** [comment]
Yes. OpenOffice supports programmatic control like ActiveX. LibreOffice too. And years ago I thought about supporting one of them as well. Complexity is in the way how to recognize which engine to use if more than one is installed (for example, MS Office and LibreOffice). And I thought about the situation when none of supported apps is installed. About that time I discovered for myself pandoc, the tool supporting many formats, and I stopped thinking in this direction.

**Comment by wesinator** [comment]
I came across pandoc recently
Yes, probably not worth implementing the office suite activex's unless you want to for a side project.
I also noticed the activex interfaces aren't installed by the default / "typical" install of those alt office suites.

Thanks,

[NO_TRAIN]::
